### PR TITLE
clipmenu: init at 5.4.0

### PR DESCRIPTION
**Pull request by jb55**
###### Motivation for this change

clipmenu is a tool for managing your clipboard with dmenu. see https://github.com/cdown/clipmenu

also add a nixos systemd user module for running clipmenud

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [xd] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/clipmenu/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ clipnotify, makeWrapper, xsel, dmenu2, utillinux, gawk, stdenv, fetchFromGitHub, lib }:
+let
+  makeBinPath = pkgs: lib.concatStringsSep ":" (map (pkg: "${lib.getBin pkg}/bin") pkgs);
```
Comment (xeji) [inline pkgs/applications/misc/clipmenu/default.nix]: Why write this yourself? Please use `lib.makeBinPath` instead.

**Comment by jb55** [comment]
> Why write this yourself? Please use `lib.makeBinPath` instead.

I knew this existed... I was searching for it and couldn't find anything... thanks xD

File: nixos/modules/services/misc/clipmenu.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    enable = mkEnableOption "clipmenu, the clipboard management daemon";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.clipmenu;
+      defaultText = "pkgs.clipmenu";
+      description = "clipmenu derivation to use.";
+    };
+
+    install = mkOption {
```
Comment (xeji) [inline nixos/modules/services/misc/clipmenu.nix]: Looks like this option's value has no effect at all. This is probably not what you want.

Comment (jb55) [reply nixos/modules/services/misc/clipmenu.nix]: oops yeah I copied it over from another module, fixing

**Pull request merged**